api/repo.go

```diff
@@ -90,87 +91,90 @@ func CreateRepo(c *gin.Context) {
 		return
 	}
 
+	// get repo information from the source
+	r, err := source.FromContext(c).GetRepo(u, input)
+	if err != nil {
+		retErr := fmt.Errorf("unable to retrieve repo info for %s from source: %w", r.GetFullName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
 	// update fields in repo object
-	input.SetUserID(u.GetID())
+	r.SetUserID(u.GetID())
 
 	if input.Active == nil {
-		input.SetActive(true)
+		r.SetActive(true)
 	}
 
 	if input.GetTimeout() == 0 {
-		input.SetTimeout(constants.BuildTimeoutMin)
+		r.SetTimeout(constants.BuildTimeoutMin)
 	}
 
 	if len(input.GetVisibility()) == 0 {
-		input.SetVisibility(constants.VisibilityPublic)
-	}
-
-	if len(input.GetFullName()) == 0 {
-		input.SetFullName(fmt.Sprintf("%s/%s", input.GetOrg(), input.GetName()))
-	}
-
-	if len(input.GetBranch()) == 0 {
-		input.SetBranch("master")
```

Comment:
These properties are are now returned via `GetRepo` as part of the successful response from the source.